extensions/src/org/pentaho/platform/web/http/api/resources/operations/CopyFilesOperation.java

```diff
@@ -262,7 +263,9 @@ private void copyRenameMode( RepositoryFile repoFile ) {
       repositoryFile =
         getRepository().createFile( destDir.getId(), duplicateFile, data, acl, null );
     }
-
+    if ( repositoryFile == null ) {
+      throw new UnifiedRepositoryAccessDeniedException( "Not enough permission to write file." );
```

Comment:
This message needs to be externalized to a properties file as do several other messages in this class.